lib/transaction.js

```diff
@@ -64,4 +65,39 @@ Transaction.manyToBuffer = function(transactions) {
   return BaseTransaction.manyToBuffer(transactions);
 };
 
+/**
+ * Override Bitcore's toObject() to include populated inputs and txid
+ */
+Transaction.prototype.toObject = function toObject() {
```

Comment:
`toObject()` will already include the output on an input if it's there: https://github.com/bitpay/bitcore/blob/master/lib/transaction/input/input.js#L74 And the hash will be include in the next release of Bitcore.